### PR TITLE
Handle gem2s response

### PR DESCRIPTION
**Pull request by cosa65**
# Background
#### Link to issue 
https://biomage.atlassian.net/browse/BIOMAGE-947
https://biomage.atlassian.net/browse/BIOMAGE-984

#### Link to staging deployment URL 
https://ui-martinfosco-ui274-api127-w.scp-staging.biomage.net/

#### Links to any Pull Requests related to this
https://github.com/biomage-ltd/ui/pull/274
https://github.com/biomage-ltd/api/pull/127

#### Anything else the reviewers should know about the changes here
Following the way we handle tasks completion in the qc pipeline, in gem2s the api needs to receive status updates when steps are finished. This way we can cause the api to send updates to the ui on what step we are at, this PR is my attempt to do this, it was working in localhost but there might be a better way to do it so feel free to comment/change the code as necessary.

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [ ] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: pipeline-runner/src/data-ingest/0-download_gem2s.r [hunk 1]
```diff
@@ -60,4 +60,6 @@ task <- function(input,pipeline_config) {
     message(exportJSON)
     write(exportJSON, "/input/meta.json")
     message('Written config json')
+
+    send_update_to_api(pipeline_config, experiment_id = input$experimentId, status_msg = "OK")
```
Comment (xverges) [inline pipeline-runner/src/data-ingest/0-download_gem2s.r:64]: Would a status message that included the task name make debugging from the API easier?

File: pipeline-runner/src/data-ingest/5_Upload-to-aws.r [hunk 2]
```diff
@@ -223,6 +212,17 @@ task <- function(input, pipeline_config) {
   cluster_env <- pipeline_config$cluster_env
   print(sprintf("Experiment ID: %s uploaded to %s.", experiment_id, cluster_env))
 
+  send_dynamodb_item_to_api(pipeline_config,
```
Comment (xverges) [inline pipeline-runner/src/data-ingest/5_Upload-to-aws.r:215]: <Rant not strictly related to this PR>
I think that we are leaking too much information from the API to the pipeline: dynamodb and table names should be something that probably belong only to the API.  But changing this is out of the scope of this PR.
</Rant not strictly related to this PR>

**Review by xverges — APPROVED** [review]
lgtm

File: pipeline-runner/src/data-ingest/5_Upload-to-aws.r (hunk 2 above)
Comment (cosa65) [reply pipeline-runner/src/data-ingest/5_Upload-to-aws.r:215]: We should, could you create a ticket for it?

**Pull request merged**